src/libs/invoke/invoke.c

```diff
@@ -337,31 +338,75 @@ void elektraInvokeClose (ElektraInvokeHandle * handle, Key * errorKey)
 }
 
 /**
- * Invokes a deferable function on a plugin handle.
+ * Invokes a deferable function on an invoke handle.
  * If the function is exported by the plugin it is directly invoked,
  * if the plugin supports deferring calls, the call is deferred.
  *
  * The parameters key set can be freed afterwards.
  *
  * @param  handle                    invoke handle
  * @param  elektraPluginFunctionName function name
- * @param  ks                        parameter key set
+ * @param  parameters                parameter key set
  * @retval 0 on success
- * @retval 1 when function not exported and deferring unsupported by plugin
+ * @retval 1 when function not exported and deferring is unsupported by plugin
  */
 int elektraInvokeCallDeferable (ElektraInvokeHandle * handle, const char * elektraPluginFunctionName, KeySet * parameters)
 {
-	ElektraDeferredCallable direct = *(ElektraDeferredCallable *) elektraInvokeGetFunction (handle, elektraPluginFunctionName);
+	if (!handle)
+	{
+		return 1;
+	}
+	return elektraDeferredCall (handle->plugin, elektraPluginFunctionName, parameters);
+}
+
+/**
+ * Execute deferred calls from list on given invoke handle.
+ *
+ * Used internally by plugins holding invoke handles.
+ *
+ * @param handle invoke handle
+ * @param list   list
+ */
+void elektraInvokeExecuteDeferredCalls (ElektraInvokeHandle * handle, ElektraDeferredCallList * list)
+{
+	if (!handle)
+	{
+		return;
+	}
+	elektraDeferredCallsExecute (handle->plugin, list);
+}
+
+/**
+ * Call a deferable function on a plugin handle.
+ * If the function is exported by the plugin it is directly invoked,
+ * if the plugin supports deferring calls, the call is deferred.
+ *
+ * The parameters key set can be freed afterwards.
+ *
+ * @param  handle                    invoke handle
+ * @param  elektraPluginFunctionName function name
+ * @param  parameters                parameter key set
+ * @retval 0 on success
+ * @retval 1 when function not exported and deferring is unsupported by plugin
```

Comment:
should be negative (like other errors). How can the user know which situation happened?

Reply:
I'm not sure if the user needs to know what happened. I think it is only interesting to know whether the call failed (cannot deferr and cannot call directly) or whether there was success (either deferred or directly called [or both]).